tasks/utils.py

```diff
@@ -376,19 +377,55 @@ def query_version(ctx, git_sha_length=7, prefix=None, major_version_hint=None):
     return version, pre, commit_number, git_sha, pipeline_id
 
 
+def cache_version(ctx, git_sha_length=7, prefix=None):
+    """
+    Generate a pickle cache file containing all needed variables used by get_version.
+    """
+    packed_data = []
+    for maj_version in ['6', '7']:
+        version, pre, commits_since_version, git_sha, pipeline_id = query_version(
+            ctx, git_sha_length, prefix, major_version_hint=maj_version
+        )
+        packed_data.append((version, pre, commits_since_version, git_sha, pipeline_id))
+    packed_data.append(is_allowed_repo_nightly_branch(os.getenv("BUCKET_BRANCH")))
+    with open("_version.cache", "wb") as file:
+        pickle.dump(packed_data, file)
+
+
 def get_version(
     ctx, include_git=False, url_safe=False, git_sha_length=7, prefix=None, major_version='7', include_pipeline_id=False
 ):
-    # we only need the git info for the non omnibus builds, omnibus includes all this information by default
-
     version = ""
-    version, pre, commits_since_version, git_sha, pipeline_id = query_version(
-        ctx, git_sha_length, prefix, major_version_hint=major_version
-    )
+    pipeline_id = os.getenv("CI_PIPELINE_ID")
+    if pipeline_id is not None:
+        try:
+            if not os.path.exists("_version.cache"):
+                os.system(
+                    f"aws s3 cp s3://dd-ci-artefacts-build-stable/{os.getenv('CI_PROJECT_NAME')}/{pipeline_id}/_version.cache . >/dev/null"
+                )
```

Comment:
Uhmm the reason I wanted to do it this way was that a lot of jobs uses ``get_version`` meaning that a lot of jobs needs to download the cache file. I want to avoid edit all of them, since it's easy to miss a few.

Reply:
My first try was with gitlab artifact but it requires that I add a dependencies for every jobs of the CI ^^'.  Same for ``before_script``, a global ``before_script`` gets overridden by a local ``before_script``of a job, ultimately needing to edit most jobs.

Reply:
If the only concern is arbitrary code execution , would using [subprocess.run](https://docs.python.org/3/library/subprocess.html#subprocess.run) be considered safe enough @hush-hush ?
Eg. instead of `os.system(f"aws s3 cp ... {os.getenv(...)}")` use `subprocess.run(["aws", "s3", "cp", "...{os.getenv(...)}"`

If I understand correctly how this works, it doesn't spawn a shell but just directly executes the command so there is no possibility of arbitrary code execution. The only possible attack is that it allows downloading a different file from the s3 bucket.

Reply:
That would be better indeed. Maybe let's do both: check that `CI_PIPELINE_ID` is a number to detect miss-configuration and use subprocess instead

Reply:
`os.system` shouldn't be used in invoke tasks in any case, we have `ctx.run` to run commands within the invoke context (which runs `subprocess.Popen` under the hood), so yes this would need to be changed.

Reply:
@hush-hush It should be better now. If you can re-check :D